### PR TITLE
Adding http.status_code as span attributes for both Apache/Nginx

### PR DESCRIPTION
**Pull request by DebajitDas**

### CODE REVIEW

**Review by kpratyus — APPROVED** [review]

File: instrumentation/otel-webserver-module/include/core/api/Payload.h [hunk 1]
```diff
@@ -47,7 +47,6 @@ class RequestPayload
     std::string flavor;
     std::string client_ip;
     long port = 80;
-    long status_code = 200;
```
Comment (gnm444) [inline instrumentation/otel-webserver-module/include/core/api/Payload.h:50]: No magic numbers here? We can find a better way right?

File: instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c [hunk 2]
```diff
@@ -1101,18 +1101,18 @@ static void stopMonitoringRequest(ngx_http_request_t* r,
     }
 
     APPD_SDK_STATUS_CODE res;
-    unsigned int errCode=0;
```
Comment (gnm444) [inline instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c:1104]: Same as above..Lets use a better practice

File: instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c [hunk 3]
```diff
@@ -1101,18 +1101,18 @@ static void stopMonitoringRequest(ngx_http_request_t* r,
...
     char* msg = NULL;
 
     if (otel_requestHasErrors(r))
     {
-        errCode=(unsigned int)otel_getErrorCode(r);
+        res_payload->status_code = (unsigned int)otel_getErrorCode(r);
         msg = (char*)malloc(6);
```
Comment (gnm444) [inline instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c:1109]: Please replace fixed number 6 by a constant

File: instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp [hunk 4]
```diff
@@ -131,7 +136,6 @@ TEST(TestRequestProcessingEngine, StartRequest)
 	payload.set_request_protocol("GET");
 	payload.set_uri("dummy_span");
 	payload.set_server_name("localhost");
-	payload.set_status_code(200);
```
Comment (gnm444) [inline instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp:134]: Same as above can't use magic numbers

File: instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp [hunk 5]
```diff
@@ -146,7 +150,6 @@ TEST(TestRequestProcessingEngine, StartRequest)
   keyValueMap[kAttrHTTPServerName] = (opentelemetry::nostd::string_view)"localhost";
   keyValueMap[kAttrHTTPMethod] = (opentelemetry::nostd::string_view)"GET";
   keyValueMap[kAttrNetHostName] =(opentelemetry::nostd::string_view)"host";
-  keyValueMap[kAttrHTTPStatusCode] = (long) 200;
```
Comment (gnm444) [inline instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp:149]: Same as above on magic numbers

File: instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp [hunk 6]
```diff
@@ -146,7 +150,6 @@ TEST(TestRequestProcessingEngine, StartRequest)
...
   keyValueMap[kAttrNETHostPort] = (long)80;
```
Comment (gnm444) [inline instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp:153]: Same as above

File: instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp [hunk 7]
```diff
@@ -229,12 +232,19 @@ TEST(TestRequestProcessingEngine, EndRequest)
 	EXPECT_CALL(*getMockSpan(interactionSpan1), End()).
 	Times(1);
 
+	unsigned int status_code = 403;
```
Comment (gnm444) [inline instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp:235]: One more magic number

File: instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp [hunk 8]
```diff
@@ -229,12 +232,19 @@ TEST(TestRequestProcessingEngine, EndRequest)
...
 	EXPECT_CALL(*getMockSpan(rootSpan), SetStatus(appd::core::sdkwrapper::StatusCode::Error, "HTTP ERROR CODE:403")).Times(1);
+	EXPECT_CALL(*getMockSpan(rootSpan),
+		AddAttribute(kAttrHTTPStatusCode,
+			HasUnsignedIntValue(status_code))).Times(1);
 	
 	EXPECT_CALL(*getMockSpan(rootSpan), End()).
 	Times(1);
 
-	auto res = engine.endRequest(rContext, "403");
```
Comment (gnm444) [inline instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp:237]: String literals should go into a resource file ..outside the code

**Review by gnm444 — CHANGES_REQUESTED** [review]

File: instrumentation/otel-webserver-module/include/core/api/Payload.h (hunk 1 above)
Comment (DebajitDas) [reply instrumentation/otel-webserver-module/include/core/api/Payload.h:50]: This is a removed code.

File: instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c (hunk 2 above)
Comment (DebajitDas) [reply instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c:1104]: This is unsigned int initialised to zero value. Do you think we really need to define this else where

File: instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp (hunk 6 above)
Comment (DebajitDas) [reply instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp:153]: This is removed code

File: instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp (hunk 5 above)
Comment (DebajitDas) [reply instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp:149]: Removed code

File: instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c (hunk 3 above)
Comment (DebajitDas) [reply instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c:1109]: Agreed

**Pull request merged**